### PR TITLE
Use correct propType for data payload

### LINKED ISSUE #3051: Some components using wrong propType for data object
**Describe the bug**
The `withData` and `withPageWrapper` PageHandlers are using an incorrect propType for the `data` prop. This means that we do not currently log console errors if invalid data is supplied.

For example, this is wrong: https://github.com/bbc/simorgh/blob/a80933a155dbb014e9a043f47a313a6b647b8438/src/app/containers/PageHandlers/withData/index.jsx#L54-L57

as `articlePropTypes` looks like this: https://github.com/bbc/simorgh/blob/30ea875c708283f71bcfd0501e85e28e256229f3/src/app/models/propTypes/article/index.js#L14-L19

and since none of the props are required, no errors are thrown for invalid data. This might also be happening in other places too.

**To Reproduce**

1. Put invalid properties into the `data` object
2. Run in development mode
3. Observe no console errors generated

**Expected behavior**
The correct propType should be used and console errors should be thrown if invalid object is given for the `data` prop.

**Additional context**
None.

### PR DESCRIPTION
**Pull request by jamesdonoh**
Resolves #3051 and #3060

**Overall change:** _Ensures that the structure of the data payload is validated correctly by using the correct propType and fixes some console errors_

NB: this change only affects console errors shown during development mode. It should not have any impact on end-users.

**Code changes:**

- Add new `data` propType file for validation of the `data` and unified `pageData` structures
- Make `model` prop inside the article data structure required (as it presumably is?)
- Fixes `checker is not a function` warnings that were being thrown for `/persian/articles/c19v65z3lg5o` due to incorrect use of `shape` (#3060)
- Migrate some existing `oneOf` data validation (Chartbeat, ATI) to use the unified propType

---

- [x] I have assigned myself to this PR and the corresponding issues
- [NA] Tests added for new features
- [NA] Test engineer approval

### CODE REVIEW

File: src/app/models/propTypes/inline/index.js [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
+export const inlineModelPropTypes = {
+  language: string.isRequired,
+  text: string.isRequired,
+  ...blocksWithTypes([fragmentBlockPropTypes]),
+};
 
-export default inlineBlockPropTypes;
+export const inlineBlockPropTypes = blockOfTypesAndModel(
+  ['inline'],
+  inlineModelPropTypes,
```
Comment (dr3) [inline src/app/models/propTypes/inline/index.js:13]:
```suggestion
  model: inlineModelPropTypes,
```
is this not a model with a shape? https://github.com/bbc/simorgh/pull/3052/files#r312831553

File: src/app/models/propTypes/inline/index.js [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 
-const inlineBlockPropTypes = shape({
-  type: 'inline',
-  model: shape({
-    text: string.isRequired,
-    blocks: arrayOf(oneOf([fragmentBlockPropTypes])),
-    language: string.isRequired,
-  }),
-});
+export const inlineModelPropTypes = {
```
Comment (dr3) [inline src/app/models/propTypes/inline/index.js:5]: Should this be a shape still?

Comment (jamesdonoh) [reply src/app/models/propTypes/inline/index.js:5]: In this context it needs to be a propType object not a `shape`, see for comparison https://github.com/bbc/simorgh/blob/3debe614a92cd9df3b900680607788a1db9adeb3/src/app/models/propTypes/inlineLink/index.js#L5-L9

Will be utilised here: https://github.com/bbc/simorgh/pull/2791/files#diff-c9811bf222651dd1e0e5f119bdd0a118R21

File: src/app/models/propTypes/inline/index.js (hunk 1 above)
Comment (jamesdonoh) [reply src/app/models/propTypes/inline/index.js:13]: This is already handled by https://github.com/bbc/simorgh/blob/9539b4a0ab5f83547c1a5612534b776c3d678152/src/app/models/propTypes/general/index.js#L6

See also e.g.
https://github.com/bbc/simorgh/blob/e549207ee04a323ce753c4786b3defccc209c5dc/src/app/models/propTypes/headline/index.js#L10

**Review by dr3 — APPROVED** [review]

**Review by chris-hinds — APPROVED** [review]
LGTM

**Comment by jamesdonoh** [comment]
This only changes propType code which is only used in development so if all the automated test suites pass the risk should be very low, so not worth manual testing. Moving to merge.

**Pull request merged**